### PR TITLE
User and role

### PR DESCRIPTION
**Pull request by Ketch92**

### CODE REVIEW

File: src/main/java/com/core/service/mapper/UserMapperImpl.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import com.core.model.dto.UserResponseDto;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class UserMapperImpl
```
Comment (YuraMitriakhin) [inline src/main/java/com/core/service/mapper/UserMapperImpl.java:14]: Do we need to create mapper and dto in first task?

**Review by YuraMitriakhin — APPROVED** [review]

File: src/main/java/com/core/service/mapper/UserMapperImpl.java (hunk 1 above)
Comment (Ketch92) [reply src/main/java/com/core/service/mapper/UserMapperImpl.java:14]: Nope, but I did

File: src/main/java/com/core/repository/RoleRepository.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.core.model.Role;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface RoleRepository extends JpaRepository<Role, Long> {
+    @Query("from Role r where r.role = :role")
+    Optional<Role> getRoleByName(@Param("role") Role.RoleTitle roleTitle);
```
Comment (boroda4436) [inline src/main/java/com/core/repository/RoleRepository.java]: `@Param` is redundant. Also, can we avoid using `@Query`?

File: src/main/java/com/core/repository/UserRepository.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("select u from User u join fetch u.rolesSet where u.phoneNumber = :number")
+    Optional<User> getUserByPhoneNumber(@Param("number") String number);
+    
+    @Query("select u from User u join fetch u.rolesSet where u.id = :id")
+    Optional<User> getById(@Param("id") Long id);
```
Comment (boroda4436) [inline src/main/java/com/core/repository/UserRepository.java]: @Ketch92 Should we override default methods?

**Review by boroda4436 — CHANGES_REQUESTED** [review]

File: src/main/java/com/core/repository/UserRepository.java (hunk 3 above)
Comment (Ketch92) [reply src/main/java/com/core/repository/UserRepository.java]: Well, that was a point of my question to you in yesterday's lecture. The user has a collection field with lazy fetching thus we having here N+1 problem. That is why I overrode the default method.  I know it's just one to 5 roles for rare use, but still. Maybe, there is a better suggestion for that?

File: src/main/java/com/core/repository/RoleRepository.java (hunk 2 above)
Comment (Ketch92) [reply src/main/java/com/core/repository/RoleRepository.java]: Hey Bohdan!) Yes, we definitely can.

**Pull request merged**